src/net_processing.cpp

```diff
@@ -4028,6 +4130,19 @@ void PeerManagerImpl::ProcessMessage(CNode& pfrom, const std::string& msg_type,
         }
 
         pfrom.fSuccessfullyConnected = true;
+
+        if (pfrom.IsPrivateBroadcastConn()) {
+            // The peer may intend to later send us NetMsgType::FEEFILTER limiting
+            // cheap transactions, but we don't wait for that and thus we may send
+            // them a transaction below their threshold. This is ok because this
+            // relay logic is designed to work even in cases when the peer drops
+            // the transaction (due to it being too cheap, or for other reasons).
+            PushPrivateBroadcastTx(pfrom);
+
+            peer->m_ping_queued = true; // Ensure a ping will be sent: mimick a request via RPC.
+            MaybeSendPing(pfrom, *peer, GetTime<std::chrono::microseconds>());
```

Comment:
What if the peer doesn't answer our ping (because they are broken / malicious)? I think we would never disconnect them because the disconnection logic is inside of `MaybeSendPing()`, which will never be called again for private peers.

Reply:
Continuing at https://github.com/bitcoin/bitcoin/pull/29415#discussion_r1790649950